geode-core/src/main/java/org/apache/geode/cache/client/internal/ExecuteFunctionOp.java

```diff
@@ -83,8 +82,8 @@ public static void execute(final PoolImpl pool,
     } else {
 
       boolean reexecute = false;
+      int maxRetryAttempts = -1;
```

Comment:
This can be a bit confusing since it's(-1) the same value as DEFAULT_RETRY_ATTEMPTS.

Would it make sense to pull out a local variable such as "NEEDS_TO_CALCULATE_RETRY_ATTEMPTS = -2?"